integration/tests/cook/test_basic.py

```diff
@@ -1064,6 +1064,11 @@ def test_list_jobs_by_pool(self):
             jobs.append(util.load_job(self.cook_url, job_uuid))
             job_uuid, resp = util.submit_job(self.cook_url, pool=pool_name, name=name, command='exit 0')
             self.assertEqual(201, resp.status_code)
+            # We wait for the job to start running, and only then start waiting for it to
+            # complete; otherwise we could get a false-negative on wait_for_job 'completed'
+            # because of a scheduling delay. Having two separate waits also disambiguates
+            # the root cause of a wait-timeout failure.
+            util.wait_for_job_in_statuses(self.cook_url, job_uuid, ['completed', 'running'])
```

Comment:
Wait for the job to start running, and only then start waiting for it to complete; otherwise we could get a false-negative on wait_for_job 'completed' because of a scheduling delay.